docs/en/shotgun/webhooks.md

```diff
@@ -0,0 +1,178 @@
+      "type": "Project"
+    },
+    "operation": "update",
+    "created_at": "2019-07-12 21:14:36.598835",
+    "event_type": "Shotgun_Asset_Change",
+    "session_uuid": "07473c00-a4ea-11e9-b3b8-0242ac110006",
+    "attribute_name": "description",
+    "event_log_entry_id": 248249
+  }
+}
+```
+
+##### Session UUID
+
+Provided as part of the event payload is the `session_uuid` that triggered the event in Shotgun. This value can be provided to [Shotgun's Python API](https://developer.shotgunsoftware.com/python-api/reference.html?highlight=session_uuid#shotgun_api3.shotgun.Shotgun.set_session_uuid), which will cause any open browser session with that session_uuid to display updates for events generated by the API.
+
+##### Payload size
+
+The maximum size of a delivery's payload is 1 megabyte. Any event triggered in Shotgun that would result in a payload size larger than 1 megabyte will have its `new_value` and `old_value` keys removed, and a `warning` key added that contains a message explaining what happened, why, and how to retrieve the entire event log entry from Shotgun.
+
+## Testing webhooks
+
+You can use any of the freely available webhook URL generators online for testing purposes. These services are specifically intended to be used for testing webhooks and other types of HTTP requests. This is a great way to get started learning about webhooks without needing to set up any infrastructure on your own network.
+
+### Using webhook.site
+
+We recommend [webhook.site](https://webhook.site). It provides a unique URL that can be copied and pasted into a webhook, and will show you deliveries made to that address in real time. The page can be customized to respond to deliveries with a specific status code and body, which means you can test delivery success and failure.
+
+The webhook.site service is aggressively rate limited. This means that it is easy to end up in a situation where some deliveries are rejected, resulting in unstable or failed webhooks. When testing, we recommend that you use a known, controllable project environment rather than live data in production. **It is also not good to send production event data to publicly available, third party web services!**
+
+### Responding to deliveries
+
+A consumer service must respond to deliveries in order for the system to consider them successfully delivered.
+
+#### Status codes
+
+| Status | Code | Description |
+|--------|:----:|:-----------:|
+| Success | < 400 | The delivery was received and processed successfully. |
+| Error | >= 400 | The delivery was received but was not processed successfully. |
+| Redirect | 3xx | The delivery was received, but should be redirected to another URL. |
+
+#### Response time
+
+**A response must be received within 6 seconds of delivery to a webhook’s URL, after which the connection will be closed.** Failure to respond in time will result in a failed delivery. 
+
+### Acknowledgement
+
+*TODO*
+
+#### What are acknowledgements used for?
+
+*TODO*
+
```

Comment:
Some general notes:

 - Is it worth mentioning Zapier, or that services like it exist, in some form here? It just seems like wildly useful information that expands what you can do quickly with webhooks. Maybe in the examples section? Or perhaps add a short, high-level section on what you can do with your payload? 
 - Once SIGGRAPH has passed and we've done the dev day webinars, we should link to Brandon's here. 
 - I think this should go above Event Driven Triggers in the TOC – it's poised as a better solution for the same types of problems, yes?
 - In fact, is it worth mentioning how it differs from event daemon and/or some thoughts on when to use which?

Reply:
- Not sure on Zapier or the like. It is useful, but it's also not something I have any experience with or content to use for. If we wanted to do that, maybe it makes sense as a separate tutorial instead of being part of the overview dev docs here?
- +1 on the Siggraph video when it's available. We'll do that as a separate task once we have it.
- I've moved it above in the TOC!
- Probably/Possibly. I'll run it by Rob/Zoe and maybe they can tell us what kind of message they want to convey for that.

Reply:
For event daemon versus webhooks, I think we should help people understand the differences and when one might be an easier choice than the other.

In general, I think Webhooks will be preferred if you have a way of responding to URL calls from Shotgun with the functionality you are trying to achieve. If you have that infrastructure (running your own server, via Amazon lambdas, via Zapier, etc.) then long term I think webhooks is a better choice.

There are a lot of possible reasons you don't have that choice, although I think most of them boil down to the fact that you can't run a server that can answer SG requests (ie on the internet) and also has access to the production resources that the code'll want to interact with (ie directory creation).

A little bit that captures the above would be nice.

Reply:
I've added a short section towards the beginning of the document outlining the differences between WH and ED. Workshopped it a few times with Rob. It's not perfect, but I think it's a decent explanation to start with.